cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -178,16 +180,20 @@ static Bom addComponent(final String compName, final String name, final String v
         bom.addComponent(comp);
         return bom;
     }
-    static Bom addComponentProperty(final String name, final String value, final String fileName) {     // Method to store Component --> Property --> name-value pairs
+    static Bom addComponentProperty(final String compName, final String name, final String value, final String fileName) {     // Method to add Component --> Property --> name-value pairs
         Bom bom = readJSONfile(fileName);
-        Property prop1 = new Property();
-        List<Property> prop = new ArrayList<>();
-        Component comp = new Component();
-        prop1.setName(name);
-        prop1.setValue(value);
-        prop.add(prop1);
-        comp.setProperties(prop);
-        bom.addProperty(prop1);
+        List<Component> componentArrayList = bom.getComponents();
+        for (Component item : componentArrayList) {
+            if (Objects.equals(item.getName(), compName)) {
```

Comment:
Use of Objects.equals() is probably unnecessary here. Typically Objects.equals() is used where an object reference can be a valid null value, as this will return true for null == null.
This can simply be:
```
item.getName().equals(compName)
```